drivers/pwm/pwm_nxp_flexio.c

```diff
@@ -0,0 +1,374 @@
+		FLEXIO_SetTimerConfig(flexio_base, child->res.timer_index[ch_id], &timerConfig);
+
+#if (defined(FSL_FEATURE_FLEXIO_HAS_PIN_REGISTER) && FSL_FEATURE_FLEXIO_HAS_PIN_REGISTER)
+		/* Reset the value driven on the corresponding pin */
+		FLEXIO_SetPinLevel(flexio_base, pwm_info->pin_id, false);
+		FLEXIO_ConfigPinOverride(flexio_base, pwm_info->pin_id, false);
+#endif
+		if (pwm_info->polarity == FLEXIO_PWM_ACTIVE_HIGH) {
+			/* Timer output is logic one and is not affected by timer reset */
+			timerConfig.timerOutput = kFLEXIO_TimerOutputOneNotAffectedByReset;
+			/* Set the timer mode to dual 8-bit counter PWM high */
+			timerConfig.timerMode = kFLEXIO_TimerModeDual8BitPWM;
+
+		} else {
+			/* Timer output is logic zero and is not affected by timer reset */
+			timerConfig.timerOutput = kFLEXIO_TimerOutputZeroNotAffectedByReset;
+			/* Set the timer mode to dual 8-bit counter PWM low */
+			timerConfig.timerMode = kFLEXIO_TimerModeDual8BitPWMLow;
+		}
+
+		/* Timer scaling factor w.r.t Flexio Clock */
+		timerConfig.timerDecrement = pwm_info->prescaler;
+
+		/* Program the PWM pulse */
+		timerConfig.timerCompare = ((uint8_t)(pwm_info->duty_cycle - 1U)) |
+			((uint8_t)(pwm_info->period - pwm_info->duty_cycle - 1U)
+			 << FLEXIO_PWM_TIMCMP_CMP_UPPER_SHIFT);
+
+		/* Configure Timer CFG and CTL bits to support PWM mode */
+		timerConfig.timerStop = kFLEXIO_TimerStopBitDisabled;
+		timerConfig.timerEnable = kFLEXIO_TimerEnabledAlways;
+		timerConfig.timerDisable = kFLEXIO_TimerDisableNever;
+		timerConfig.timerStart = kFLEXIO_TimerStartBitDisabled;
+		timerConfig.timerReset = kFLEXIO_TimerResetNever;
+		timerConfig.triggerSource = kFLEXIO_TimerTriggerSourceInternal;
+
+		/* Enable the pin out for the selected timer */
+		timerConfig.pinConfig = FLEXIO_PWM_TIMER_PIN_OUTPUT_ENABLE;
+		timerConfig.pinPolarity = pwm_info->polarity;
+
+		/* Select the pin that the selected timer will output the signal on */
+		timerConfig.pinSelect = pwm_info->pin_id;
+
+		FLEXIO_SetTimerConfig(flexio_base, child->res.timer_index[ch_id], &timerConfig);
+	}
+
+	return 0;
+}
+
+static const struct pwm_driver_api pwm_nxp_flexio_driver_api = {
+	.set_cycles = pwm_nxp_flexio_set_cycles,
+	.get_cycles_per_sec = pwm_nxp_flexio_get_cycles_per_sec,
+};
+
+#define _FLEXIO_PWM_PULSE_GEN_CONFIG(n)								\
+	{											\
+		.pin_id = DT_PROP(n, pin_id),							\
+		.prescaler = DT_PROP(n, prescaler),						\
+		.period = DT_PROP(n, period),							\
+		.duty_cycle = DT_PROP(n, duty_cycle),						\
+		.polarity = DT_PROP(n, polarity),						\
+	},
+
+#define FLEXIO_PWM_PULSE_GEN_CONFIG(n)								\
+	pwm_nxp_flexio_channel_config_type flexio_pwm_##n##_init[] = {				\
+		DT_INST_FOREACH_CHILD_STATUS_OKAY(n, _FLEXIO_PWM_PULSE_GEN_CONFIG)		\
+	};											\
+	const struct pwm_nxp_flexio_pulse_info flexio_pwm_##n##_info = {			\
+		.pwm_pulse_channels = ARRAY_SIZE(flexio_pwm_##n##_init),			\
+		.pwm_info = (pwm_nxp_flexio_channel_config_type *)flexio_pwm_##n##_init,	\
```

Comment:
nit: this typecasting is redundant.

Reply:
If I remove this typecasting the compiler says-
warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  328 |                 .pwm_info = flexio_pwm_##n##_init,

Reply:
removed !